### PR TITLE
Use bincode to store the tasks in the task queue

### PR DESCRIPTION
**Pull request by irevoire**
Mitigate but does not fix the task queue issue https://github.com/meilisearch/meilisearch/issues/3622

This is one of the multiple « fixes/mitigation » we can work on to reduce the issue of the task queue being full.

Here are some measures of the impact of moving to bincode
| db size/branch              | main                              | bincode                     | Percentage change |
|-----------------------|------------------------|---------------------|--------------------|
| db size for 100k tasks |               80MiB               |        24MiB                |    -70%                    |
| db size for 1M tasks    |              506MiB              |        100MiB              |    -79%                     |
| time to import 1M tasks |           3m36s                 |         3m37s             |         0%                   |
| db size for 10M tasks    |              4.9GiB               |        987MiB             |        -80%                |
| time to import 10M tasks |           1m05s                 |         55s                 |         0%                   |

As we can see, it roughly divides the size of the task database by a factor of 5.

### CODE REVIEW

File: index-scheduler/Cargo.toml [hunk 1]
```diff
@@ -12,7 +12,7 @@ license.workspace = true
 
 [dependencies]
 anyhow = "1.0.64"
-bincode = "1.3.3"
+bincode = "2.0.0-rc.2"
```
Comment (Kerollmops) [inline index-scheduler/Cargo.toml:15]: We will merge this PR once the `rc`s are done and released as a final version. What do you think?
They just published a `rc.3` recently on the 30th of March.

File: index-scheduler/src/utils.rs [hunk 2]
```diff
@@ -572,3 +574,30 @@ pub fn dichotomic_search(start_point: usize, mut is_good: impl FnMut(usize) -> b
         }
     }
 }
+
+pub struct Bincode<T>(std::marker::PhantomData<T>);
```
Comment (Kerollmops) [inline index-scheduler/src/utils.rs:578]: Can you add more documentation, please?

File: milli/Cargo.toml [hunk 3]
```diff
@@ -13,7 +13,8 @@ license.workspace = true
 
 [dependencies]
 bimap = { version = "0.6.2", features = ["serde"] }
-bincode = "1.3.3"
+bincodev1 = { package = "bincode", version = "1.3.3" }
```
Comment (Kerollmops) [inline milli/Cargo.toml:16]: Why do we need `bincodev1`?

**Review by Kerollmops — CHANGES_REQUESTED** [review]

File: index-scheduler/src/utils.rs (hunk 2 above)
Comment (irevoire) [reply index-scheduler/src/utils.rs:578]: Same, we won’t need this if we wait until bincodev2 is officially released

**Comment by irevoire** [comment]
We’ll wait until bincodev2 is officially released before integrating it into the codebase.

**Pull request closed without merging**